odl/tomo/data/uncompr_bin.py

```diff
@@ -0,0 +1,472 @@
+
+    return tuple(conv_list)
+
+
+class FileReaderUncompressedBinary(object):
+
+    """Reader for uncompressed binary files including a header."""
+
+    def __init__(self, file, header_fields):
+        """Initialize a new instance.
+
+        Parameters
+        ----------
+        file : file-like or str
+            Stream or filename from which to read the data. The stream
+            is allowed to be already opened in 'rb' mode.
+        header_fields : sequence of dicts
+            Definition of the fields in the header (per row), each
+            containing key-value pairs for the following keys:
+
+            - ``'name'`` : Label for the field.
+            - ``'offset_bytes'`` : Start of the field in bytes.
+            - ``'size_bytes'`` : Size of the field in bytes.
+            - ``'dtype'`` : Data type in Numpy- or Numpy-readable format.
+            - ``'dshape'`` (optional) : The array of values is reshaped to
+              this shape.
+            - ``'description'`` : A human-readable description of the field.
+        """
+        try:
+            f = open(file, 'rb')
+        except TypeError:
+            f = file
+
+        if f.mode != 'rb':
+            raise ValueError("`file` must be opened in 'rb' mode, but mode "
+                             "'is {}'".format(f.mode))
+
+        self.file = f
+
+        # Initialize some attributes to default values, plus the header
+        self.data_shape = -1  # Makes reshape a no-op
+        self.header_bytes = None
+        self.header_fields = header_fields
+        self.data = None
+        if self.header_fields:
+            self.header = self.read_header()
+        else:
+            self.header = None
+
+    @classmethod
+    def from_raw_file(cls, file, header_bytes, dtype):
+        """Construct a reader from a raw file w/o header spec.
+
+        Readers constructed with this method can use `read_data`, but
+        not `read_header`.
+
+        Parameters
+        ----------
+        file : file-like or str
+            Stream or filename from which to read the data. The stream
+            is allowed to be already opened in 'rb' mode.
+        header_bytes : int
+            Size of the header in bytes.
+        dtype :
+            Data type specifier for the data field. It must be
+            understood by the `numpy.dtype` constructor.
+
+        Returns
+        -------
+        reader : `MRCFileReader`
+            Raw reader for the given MRC file.
+        """
+        try:
+            f = open(file, 'rb')
+        except TypeError:
+            f = file
+
+        if f.mode != 'rb':
+            raise ValueError("`file` must be opened in 'rb' mode, got '{}'"
+                             "".format(f.mode))
+
+        header_bytes = int(header_bytes)
+        if header_bytes < 0:
+            raise ValueError('`header_bytes` must be nonnegative, got {}.'
+                             ''.format(header_bytes))
+
+        filesize_bytes = f.seek(0, 2)
```

Comment:
The 2 means "backwards from the end of the file". Unfortunately it cannot be used as keyword argument, so a comment will have to explain this.